ui/test/nuxeo-video-conversions.test.js

```diff
@@ -0,0 +1,56 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { fixture, flush, html, isElementVisible } from '@nuxeo/nuxeo-elements/test/test-helpers';
+import '../nuxeo-video/nuxeo-video-conversions.js';
+import videoProperties from './resources/videoProperties';
+/* eslint-disable no-unused-expressions */
+
+suite('nuxeo-video-conversions', () => {
+  suite('When no document provided', () => {
+    test('Should still display the label', async () => {
+      const element = await fixture(html`
+        <nuxeo-video-conversions role="widget" label="video conversions"></nuxeo-video-conversions>
+      `);
+      await flush();
+      expect(element.root.querySelector('h3').innerText).to.equal('video conversions');
+    });
+
+    test('Should not display any other element', async () => {
+      const element = await fixture(html`
+        <nuxeo-video-conversions role="widget" label="video conversions"></nuxeo-video-conversions>
+      `);
+      await flush();
+      expect(isElementVisible(element.shadowRoot.querySelector('.flex'))).to.be.false;
+    });
+  });
+
+  suite('When document with vid:transcodedVideos properties provided', () => {
+    test('It should display the template if the video contains conversion content', async () => {
+      const document = {
+        properties: {
+          'vid:transcodedVideos': videoProperties['vid:transcodedVideos'],
+        },
+      };
+      const element = await fixture(html`
+        <nuxeo-video-conversions role="widget" .document="${document}" label="video conversions">
+        </nuxeo-video-conversions>
+      `);
+      await flush();
+      expect(element.shadowRoot.querySelectorAll('.flex').length).to.equal(3);
```

Comment:
Also, it would be cool to try to use, as much as possible ChaiJS BDD API (for consistency purposes).

In this specific case we could have something like: `expect(element.shadowRoot.querySelectorAll('.flex')).to.have.lengthOf(3);` or even `expect(element.shadowRoot.querySelectorAll('.flex')).to.be.an('array').with.lengthOf(3);` (maybe not applicable in this specific example since querySelectorAll returns NodeList, but you got the idea 🙂 ) 

More details: https://www.chaijs.com/api/bdd/